RecoverCRPVM/RecoverCRPVM.ps1

```diff
@@ -0,0 +1,175 @@
+cd $RunPath
+$CommonFunctions = $runPath+"\Common-Functions.psm1"
+
+$CommonFunctions = $runPath+"\Common-Functions.psm1"
+
+if (Get-Module Common-Functions) {remove-module -name Common-Functions}   
+Import-Module -Name $CommonFunctions  -ArgumentList $LogFile -ErrorAction Stop 
+
+if (-not (Get-AzureRmContext).Account)
+{
+    Login-AzureRmAccount
+}
+write-log "Info: Log is being written to ==> $LogFile" 
+#Set the context to the correct subid
+Write-Log "Setting the context to SubID $SubID" -color Yellow
+$subContext= Set-AzureRmContext -Subscription $SubID
+if ($subContext -eq $null) 
+{
+    Write-Log "Unable to set the Context for the given subId ==> $SubID, Please make sure you first  run the command ==> Login-AzureRMAccount" -Color Red
+    return
+}
+
+#Step 1 Get the VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$ResourceGroup`" -Name `"$VmName`"" -Color Yellow
+$vm = Get-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName 
+if (-not $vm)
+{
+    Write-Log "Unable to find the VM `"$VmName`",  cannot proceed, please verify the VM name and the resource group name." -Color Red
+    return
+}
+Write-Log "Successfully got the VM Object info for ==> $($vm.Name)" -Color Green
+
+if ($vm.StorageProfile.OsDisk.OsType -eq "Windows") {$windowsVM= $true} else {$windowsVM= $false}
+
+$Vmname = $vm.Name
+$rescueVMNname = "$prefix$Vmname"
+$RescueResourceGroup = "$prefix$ResourceGroup"
+
+#Step 2 Get the Rescue VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$RescueResourceGroup `" -Name `"$rescueVMNname`"" -Color Yellow
+$rescuevm = Get-AzureRmVM -ResourceGroupName $RescueResourceGroup -Name $rescueVMNname
+if (-not $rescuevm)
+{
+    Write-Log "Unable to find the VM `"$rescueVMNname`",  cannot proceed, please verify the VM name and the resource group name." -Color Red
+    return
+}
+Write-Log "Successfully got the VM Object info for the Rescue VM ==>  $rescueVMNname" -Color Green
+
+#Step 3 -Removing the DataDisk from Rescue VM
+$VHDNameShort = ($FixedOsDiskUri.Split('/')[-1]).split('.')[0]
+Write-Log "Removing the Data disk from the Rescue VM ==>  $rescueVMNname" -Color Yellow
+Remove-AzureRmVMDataDisk -VM $rescuevm -Name $VHDNameShort
+Update-AzureRmVM -ResourceGroupName $RescueResourceGroup -VM $rescuevm
+Write-Log "Successfully removed the Data disk from the Rescue VM ==>  $rescueVMNname" -Color Green
+
+#Stop the VM before performing the disk swap.
+Write-Log "Stopping the VM ==> $VmName"  -Color Yellow
+$stopped= Stop-AzureRmVM -ResourceGroupName $ResourceGroup -Name $VmName
+if ($Stopped)
+{
+   Write-Log "Successfully stopped the  VM ==> $VmName" -Color Green
+}
+
+
+#Step 4 -Disk Swapping the OS Disk to point to the fixed OSDisk Uri
+
+Write-Log "Disk Swapping the OS Disk, to point to the fixed OS Disk for VM ==>  $VmName" -Color Yellow
+#before setting the uri, ensure to remove any new line characters
+$FixedOsDiskUri  = $FixedOsDiskUri.Replace("`r`n","")
```

Comment:
Lo-pri: Might be good idea to do this before the detach when trying to figure out vhd name